LSPS3/README.md

```diff
@@ -0,0 +1,317 @@
+|Version | 1                        |
+
+## Motivation
+
+A "JIT Channel" is a 0-conf channel opened in response to an incoming payment
+from the public network to a client, via the LSP.
+This allows a client with no Lightning channels to start receiving on
+Lightning, and have the cost of their inbound liquidity be deducted from their
+first received payment.
+
+### Trust Models
+
+As of this version, the trust model is that the LSP trusts clients to actually
+claim their payment once the channel is opened and the payment is forwarded.
+
+If the client does not claim the payment, and the funding transaction confirms,
+then the LSP will have its funds locked in a channel that was not paid for.
+Even if the LSP disables all other use of the channel until the payment paying
+for the channel is claimed, the client may then refuse all mutual close attempts
+in retaliation (by disconnecting), and force the LSP to unilaterally close,
+which locks the funds until the `to_self_delay` (indicated by the client, and
+imposed on the LSP) has passed.
+
+LSPs that implement this API SHOULD include mitigations to allow them to
+determine if clients can be trusted, or to detect and protect against attacks
+on the LSP.
+How those mitigations are performed is beyond the scope of this specification.
+
+### Amounts
+
+All amounts in this protocol are specified in millisatoshi.
+
+All millisatoshi values MUST be represented as a string and MUST NOT be
+represented as integer values.
+
+> **Rationale** JSON numbers are specified to only fit IEEE double-precision
+> numbers, which have a 53 bit mantissa.
+> The maximum number of millisatoshis (21 million BTC) cannot be specified in
+> 53 bits.
+> In addition, actual JSON parsers may use a smaller size for parsed integers;
+> for instance, as signed 32 bit numbers.
+> The largest safe value is 2,147,483,647; 2,147,483,647msat = BTC0.021,474,836,47
+> which is too low.
+
+### Actors
+
+'LSP' is the API provider.
+'Client' is the API client.
+'Payer' is who pays for the initial receive of the client.
+
+## Flow (LSP Trusts Client Model)
+
+Overview:
+
+* The client determines that it would like to receive over Lightning, but
+  it has no inbound capacity, and the client cannot pre-buy this capacity
+  (for instance, it also has no outbound capacity or onchain funds).
+* The client determines the parameters of a particular LSP via a `GET`
+  request.
+* The client requests for a JIT channel, specifying how much it will
+  receive, via a `POST` request.
+* The LSP returns an SCID, which identifies this request to the LSP.
+* The client generates an invoice, which includes the above SCID and the
+  LSP node ID as a routehint.
+* The client hands the invoice to whoever it will receive funds from.
+* The payment is forwarded to the LSP.
+* The LSP recognizes the next hop SCID as being a JIT channel request,
+  and opens a 0-confirmation channel to the client, which must be
+  connected to the LSP at that time.
```

Comment:
Should the spec speak to an optional or required timeout (in case the client is offline, which could very well be the case for mobile and web wallets using an LSP)?

Reply:
There is an `opening_fee_valid_until` field the LSP provides, which is the timeout.